pkg/endpointslice/map.go

```diff
@@ -142,7 +162,18 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 	for _, endpoint := range es.Endpoints {
 		var records []serviceimport.DNSRecord
 
-		for _, address := range endpoint.Addresses {
+		addresses := endpoint.Addresses
+		if localSlice != nil {
+			for _, localEndpoint := range localSlice.Endpoints {
+				if *localEndpoint.NodeName == *endpoint.NodeName &&
```

Comment:
Not sure if we need this check. These would be same for non-globalnet while for globalnet localslice we fetched here will be more up-to-date and recent than the one we got from event notification.

Reply:
I found that the `localSlice` didn't include the hostname which is why I attempt to map back to the globalnet endpoint slice